test/forward_backward_compatibility/check_forward_backward_compatibility.py

```diff
@@ -26,111 +32,84 @@
-    ("aten::miopen_depthwise_convolution_backward", datetime.date(9999, 1, 1)),
-    ("aten::miopen_depthwise_convolution_backward_input", datetime.date(9999, 1, 1)),
-    ("aten::miopen_depthwise_convolution_backward_weight", datetime.date(9999, 1, 1)),
-    ("caffe2::", datetime.date(2021, 10, 23)),
-    ("prepacked::unpack_prepacked_sizes_conv2d", datetime.date(9999, 1, 1)),
-    ("prepacked::unpack_prepacked_sizes_linear", datetime.date(9999, 1, 1)),
-    ("q::_FloatToBfloat16Quantized", datetime.date(2021, 12, 21)),
-    ("q::_Bfloat16QuantizedToFloat", datetime.date(2021, 12, 21)),
-    ("aten::_inverse_helper", datetime.date(2021, 12, 31)),
-    ("aten::softplus_backward", datetime.date(2022, 1, 31)),
-    ("aten::softplus_backward.grad_input", datetime.date(2022, 1, 31)),
-    ("aten::quantile", datetime.date(2022, 9, 30)),
-    ("aten::nanquantile", datetime.date(2022, 9, 30)),
-    ("aten::_convolution_double_backward", datetime.date(2022, 3, 31)),
-    ("aten::_scatter_reduce", datetime.date(2022, 1, 31)),
-    ("aten::native_multi_head_self_attention", datetime.date(9999, 1, 1)),
-    ("aten::_native_multi_head_self_attention", datetime.date(9999, 1, 1)),
+    "profiler::_call_end_callbacks_on_jit_fut*",
+    "profiler::_record_function_enter",
+    "aten::_cholesky_helper",
+    "aten::_lstsq_helper",
+    "aten::_syevd_helper",
+    "aten::_linalg_solve_out_helper_",
+    "aten::select_backward",
+    "aten::slice_backward",
+    "aten::diagonal_backward",
+    "aten::rowwise_prune",
+    "aten::adaptive_avg_pool3d_backward",
+    "aten::_embedding_bag_dense_backward",
+    "aten::randperm",
+    "aten::_convolution_nogroup",
+    "aten::miopen_convolution_backward",
+    "aten::miopen_convolution_backward_bias",
+    "aten::miopen_convolution_backward_input",
+    "aten::miopen_convolution_backward_weight",
+    "aten::miopen_convolution_transpose_backward",
+    "aten::miopen_convolution_transpose_backward_input",
+    "aten::miopen_convolution_transpose_backward_weight",
+    "aten::miopen_depthwise_convolution_backward",
+    "aten::miopen_depthwise_convolution_backward_input",
+    "aten::miopen_depthwise_convolution_backward_weight",
+    "prepacked::unpack_prepacked_sizes_conv2d",
+    "prepacked::unpack_prepacked_sizes_linear",
+    "aten::native_multi_head_self_attention",
+    "aten::_native_multi_head_self_attention",
 ]
 
-ALLOW_LIST_COMPILED = [
-    (
-        re.compile(item[0]),
-        item[1],
-        re.compile(item[2]) if len(item) > 2 else None,
-    ) for item in ALLOW_LIST if item[1] >= datetime.date.today()
-]
+def compile_temp_allow_list():
+    output = []
+    for item in TEMPORARY_ALLOW_LIST:
+        deadline = item[1]
+        today = datetime.date.today()
+        interval = deadline - today
+        if deadline > today and deadline - today < MAX_ALLOWED_PERIOD:
+            output.append((re.compile(item[0]), deadline, re.compile(item[2]) if len(item) > 2 else None))
```

Comment:
What is `item[2]`?

Reply:
This was here before. (https://github.com/pytorch/pytorch/pull/72474/files#diff-236fbde71e59cb1597cac177a83e49fb62b30770eec55c4e7a0f2650b9eb6203L25) And this is pretty much a copy of what we had before. But now I think about it, it is probably ok to delete it